src/main/java/org/mamute/auth/LdapSSOFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.mamute.auth;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import org.mamute.dao.UserDAO;
+import org.mamute.model.LoggedUser;
+import org.mamute.model.User;
+
+import br.com.caelum.vraptor.environment.Environment;
+import br.com.caelum.vraptor.events.MethodReady;
+
+public class LdapSSOFilter {
+
+	@Inject private Environment env;
+	@Inject private UserDAO users;
+	@Inject private LDAPApi ldap;
+	@Inject private LoggedUser loggedUser;
+	@Inject private HttpServletRequest request;
+	@Inject private Access system;
+
+	public void checkSSO(@Observes MethodReady methodReady) {
+		if (env.supports(LDAPApi.LDAP_SSO) && !loggedUser.isLoggedIn() && request.getRequestURI() != null
+				&& !request.getRequestURI().endsWith("/logout")) {
+			String userName = request.getRemoteUser();
```

Comment:
I'm still not convinced that such scenarios should / must be covered in every application wanting to offer SSO. There are solutions for mapping data into the standard remoteUser, see e.g. here: http://stackoverflow.com/questions/7553967/getting-a-value-from-httpservletrequest-getremoteuser-in-tomcat-without-modify/
This uses a Valve in Tomcat, and this would allow you to provide a remote user from any data source (e.g. a custom header).

Similar solutions should exist for other application servers.